connect/tests/src/test/java/com/telenor/connect/utils/IdTokenValidatorTest.java

```diff
@@ -146,7 +149,71 @@ public void missingIssueTimeThrows() throws Exception {
         signedJWT.sign(new ECDSASigner(new BigInteger("123")));
         IdToken idToken = new IdToken(signedJWT.serialize());
 
-        IdTokenValidator.validate(idToken);
+        IdTokenValidator.validate(idToken, null);
+    }
+
+    @Test
+    public void isValidExpirationTimeReturnsFalseOnNullExpDate() {
+        boolean actual = IdTokenValidator.isValidExpirationTime(null, new Date(), null);
+        assertThat(actual, is(false));
+    }
+
+    @Test
+    public void isValidExpirationTimeReturnsTrueWhenExpDateIsAfterCurrentDate() {
+        Calendar calendar = Calendar.getInstance();
+        Date localTime = calendar.getTime();
+        calendar.add(Calendar.HOUR, 1);
+        Date future = calendar.getTime();
+
+        boolean actual = IdTokenValidator.isValidExpirationTime(future, localTime, null);
+        assertThat(actual, is(true));
+    }
+
+    @Test
+    public void isValidExpirationTimeReturnsFalseWhenCurrentTimeIsAfterExpAndServerTimestampIsMissing() {
+        Calendar calendar = Calendar.getInstance();
+        calendar.add(Calendar.HOUR, 1);
+        Date localTime = calendar.getTime();
```

Comment:
Consider naming this variable "future" to increase readability. Consider setting up "now" and "future" in the same way as in the previous test